### PR TITLE
Use requests module to download files

### PR DESCRIPTION
**Pull request by gasinvein**
requests module is easier to use, has builtin debug logging and it provides a builtin mechanism for retries, so we can get rid of `tenacity` dependency (we already depend on `requests` through `PyGithub`).

This PR shouldn't change behavior (except for default User-Agent change), just some internal work. But since it's more of a personal preference, asking for @wjt and @andrunko consent.

### CODE REVIEW

File: src/lib/utils.py [hunk 1]
```diff
@@ -36,14 +36,8 @@
 
 from collections import OrderedDict
 from ruamel.yaml import YAML
-from tenacity import (
```
Comment (wjt) [inline src/lib/utils.py:39]: Can we drop a dependency on tenacity?

**Review by wjt — COMMENTED** [review]
yes plz

File: src/lib/utils.py (hunk 1 above)
Comment (gasinvein) [reply src/lib/utils.py:39]: You mean drop it from docker image?

Comment (wjt) [reply src/lib/utils.py:39]: Yes

Comment (gasinvein) [reply src/lib/utils.py:39]: Dropped.

**Review by wjt — APPROVED** [review]

**Pull request merged**